drv/stm32xx-i2c-server/src/main.rs

```diff
@@ -296,6 +296,44 @@ fn reset_if_needed(
     }
 }
 
+///
+/// A variant of [`reset_if_needed`] that will also wiggle the SCL lines
+/// via [`wiggle_scl`].
+///
+fn reset_and_wiggle_if_needed(
+    code: ResponseCode,
+    controller: &I2cController<'_>,
+    port: PortIndex,
+    muxes: &[I2cMux<'_>],
+    muxmap: &mut MuxMap,
+    pins: &[I2cPins],
+) {
+    if reset_needed(code) {
+        let sys = SYS.get_task_id();
+        let sys = Sys::from(sys);
+
+        for pin in pins
+            .iter()
+            .filter(|p| p.controller == controller.controller)
+            .filter(|p| p.port == port)
+        {
+            wiggle_scl(&sys, pin.scl, pin.sda);
+
+            for gpio_pin in &[pin.scl, pin.sda] {
+                sys.gpio_configure_alternate(
+                    *gpio_pin,
+                    OutputType::OpenDrain,
+                    Speed::Low,
+                    Pull::None,
+                    pin.function,
+                );
+            }
```

Comment:
This is necessary because `wiggle_scl` puts the SCL pin in GPIO output mode, right? It _might_ be nice if there was a comment --- I was a bit surprised that `wiggle_scl` doesn't just put the pin back the way it found it when it's done, but I see that this is because it previously only happened during initial configuration and the pin AF configuration happens in a separate step.